File: app/src/main/java/im/tox/antox/utils/FriendInfo.java
package im.tox.antox.utils;

import java.sql.Timestamp;
import java.util.Comparator;

public class FriendInfo extends Friend{

    public int unreadCount;
    public String lastMessage;
    public Timestamp lastMessageTimestamp;

	public FriendInfo() {
		super();
	}

	public FriendInfo(boolean isOnline, String friendName, String userStatus, String userNote, String key, String lastMessage, Timestamp lastMessageTimestamp, int unreadCount, String a) {
		super(isOnline, friendName, userStatus, userNote, key, a);
        this.lastMessage = lastMessage;
        this.lastMessageTimestamp = lastMessageTimestamp;
        this.unreadCount = unreadCount;
	}
}

File: app/src/main/scala/im/tox/antox/utils/FriendInfo.scala
package im.tox.antox.utils

import java.sql.Timestamp
//remove if not needed

class FriendInfo(
  isOnline: Boolean,
  friendName: String,
  userStatus: String,
  personalNote: String,
  friendKey: String,
  val lastMessage: String,
  val lastMessageTimestamp: Timestamp,
  val unreadCount: Int,
  alias: String) extends Friend(isOnline, friendName, userStatus, personalNote, friendKey, alias) {

  /**
  Returns 'alias' if it has been set, otherwise returns 'name'.
   */
  def getAliasOrName(): String = {
    if (alias != "") alias else name
  }
}
